netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -97,10 +105,14 @@
   private final TransportTracer.Factory transportTracerFactory;
   private final InternalChannelz channelz;
   // Only modified in event loop but safe to read any time.
-  private volatile InternalInstrumented<SocketStats> listenSocketStats;
+  private volatile List<InternalInstrumented<SocketStats>> listenSocketStatsList =
```

Comment:
It can be modified in different channels' eventloop, and there is only a single one, so it looks fine at write. And the socketStats can be transient, so we return a snapshot on the direct read. I see the `getListenSocketStatsList` returned the reference directly which might cause problems as it is mutable, we can return an immutable list, or create a new copy then return.

Reply:
Actually each server channel's eventloop can be different.